base/nullable.jl

```diff
@@ -3,6 +3,27 @@
 immutable NullException <: Exception
 end
 
+"""
+    Nullable(x, hasvalue::Bool=true)
+
+When `hasvalue` is true, wrap value `x` in an object of type `Nullable`, which indicates
```

Comment:
>I think I would phrase it slightly differently: as the original docstring said (even though it was incomplete), x is always wrapped in the resulting Nullable. The hasvalue argument will determine whether the value will be considered present or not, but it's actually always there in the value field of the result (cf. #16923).

I appreciate the explanation and reference. I understand the behavior better now.

>The hasvalue argument is useful only in quite specific cases, so I'd mention it only in a separate paragraph, to keep the explanation for the one-argument form simple. The fist paragraph could almost be kept as it is currently.

Now the `hasvalue` argument is addressed in the second paragraph of the docstring only. I've included an additional doctest to better illustrate behavior when `hasvalue` is `false`.

Thanks!